builtin/plugins/coin/coin.go

```diff
@@ -213,6 +249,25 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
 	return ctx.Set(economyKey, econ)
 }
 
+//Mint : to be called by CoinDeflationManager Responsible to mint coins as per various parameter defined
+func Mint(ctx contract.Context) error {
+	var policy Policy
+	err := ctx.Get(policyKey, &policy)
+	if err != nil {
+		return errUtil.Wrap(err, "Failed to Get PolicyInfo")
+	}
+	var depreciation *common.BigUInt
+	policyNumerator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorNumerator))
+	policyDenominator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorDenominator))
+	blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
+	depreciation.Mul(policyNumerator, blockHeight)
+	depreciation.Div(depreciation, policyDenominator)
+	//amount := policy.BaseMintingAmount
```

Comment:
Why did you comment out this? 

I said only mint, if `amount` is greater than `depreciation`.

Reply:
It is a temp change for code to compile as Policy data type was modified :), mukul will complete it in short while.

Reply:
@ParthDesai he must have done it to compile it, I am completing it